### ISSUE TITLE
Figwheel sidecar causes error with cider-nrepl 0.9.0-snapshot

### ISSUE THREAD

**Issue by sktoiva** [bhauman/lein-figwheel#149]
Sidecar has in its project.clj a dependency for cider-nrepl 0.8.2:
https://github.com/bhauman/lein-figwheel/blob/master/sidecar/project.clj#L23

This will override profiles.clj plugin definition and cause a warning in cider repl if the versions differ:
```
Please, install (or update) cider-nrepl 0.9.0-SNAPSHOT and restart CIDER 
WARNING: CIDER's version (0.9.0-snapshot) does not match cider-nrepl's version (0.8.2)
```

This can be fixed with adding cider-nrepl exclusion to plugins definition in project.clj:
```clojure
[lein-figwheel "0.3.1" :exclusions [cider/cider-nrepl]]
```

As I'm currently trying to learn clojure and the best practices, I'm wondering if this dependency in sidecar should be there as it is a dependency to a development tool? And moreover, is there something that can be done on the figwheel's side?

**Comment by bhauman** [comment]
I'm still waiting for this cljs nrepl stuff to actually settle in and start working.  

Sidecar is a development tool and is currently doing a bit of a trade off to allow folks to use nrepl to connect to the running clojurescript compilation process.  I threw in cider because most people want it and I'm starting an nrepl server in process where folks can't do their own configuration.  So for now the exclusion is the way to go.

**Comment by sktoiva** [comment]
Ok, an informed decision then. 

Would it be beneficial to add that exclusion line to the README as a guidance so this wouldn't cause grey hair to people encountering the warning?

**Comment by bhauman** [comment]
Fixed this in upcoming 0.3.4 release. The cider-nrepl middleware only gets included if you have cider-nrepl available.  This way you can just add the dependency you prefer :)
